sentry-core/src/main/java/io/sentry/core/SentryCommonOptions.java

```diff
@@ -0,0 +1,263 @@
+package io.sentry.core;
+
+import com.jakewharton.nopen.annotation.Open;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Subset of {@link SentryOptions} properties used in 3rd party framework integrations. */
+@Open
+public class SentryCommonOptions {
+
+  /** @see SentryOptions#dsn */
+  private String dsn = "";
+
+  /** @see SentryOptions#shutdownTimeoutMillis */
+  private Long shutdownTimeoutMillis;
+
+  /** @see SentryOptions#flushTimeoutMillis */
+  private Long flushTimeoutMillis;
+
+  /** @see SentryOptions#readTimeoutMillis */
+  private Integer readTimeoutMillis;
+
+  /** @see SentryOptions#bypassSecurity */
+  private Boolean bypassSecurity;
+
+  /** @see SentryOptions#debug */
+  private Boolean debug;
```

Comment:
should we use primitive types? otherwise, we'd need to do NPE checks

Reply:
The advantage is being able to know if the value was set explicitly by the user and being able to flip the default after invoking a user callback.

But for this field I agree probably doesn't make sense

Reply:
As @bruno-garcia said, we need to know which values were set explicitly by user. This way we don't need to keep in sync default values on `SentryOptions` and `SentryCommonOptions`. This also includes data type for fields - they have to be nullable otherwise we loose this information.